### ISSUE TITLE
Case search | Enhance to be able to search by docket number

### ISSUE THREAD

**Issue by laurjpeterson** [department-of-veterans-affairs/caseflow#11440]
Allow users to search for Veterans' cases by docket number.

## AC
- users can search by docket number for both AMA and legacy appeals

**Comment by jcq** [comment]
I assume we would want to support this through the single existing search box, correct? What would we want to do with the regex checks for valid entries? Currently we verify that input is valid client side by checking if it is a number between 7–9 digits long (for SSN or file number). However, my understanding of docket number is that it could be anything over 7 digits (6 digits for date + unique numeric ID for appeal).

We could obviously change the regex to allow numbers of 7+ digits, but that makes it far less useful for detecting incorrect values for SSN, etc.

**Comment by leikkisa** [comment]
JC and I talked in Slack but updating the ticket also, in case there are any differing opinions out there.

> I assume we would want to support this through the single existing search box, correct?
We are assuming "yes"

> What would we want to do with the regex checks for valid entries?
If this is in the front end maybe the hyphen can be a differentiator.

> And this doesn't even account for legacy cases, which could apparently have docket numbers with entirely different formats.
We are considering this out of scope for this ticket since it says AMA appeals.

**Comment by laurjpeterson** [comment]
Correct! The initial intent of this ticket was to enhance by docket number for AMA appeals using the existing design (search box). The hyphen makes sense to me. Users can use VACOLS to search by docket number for legacy appeals. 

We can create a design backlog ticket for a future when we can search for legacy docket appeals, because it sounds like we need their input on whether this should be another search box or not. Since designers are focused on other priorities, I’m ok deferring that until later. Cc: @carodew

**Comment by jcq** [comment]
Sounds good! Restricting to AMA ones definitely makes things easier. We'll have to do more than simply rely on presence of hyphen for validation since SSN routinely could be entered that way (the current code allows hyphens and other characters to be entered, but simply removes all non-numeric chars in the request to the backend), but I'm sure we can come up with something.

**Comment by carodew** [comment]
I'm willing to take a look if someone can point me to the right place in Caseflow demo (is it this screen? http://www.caseflowdemo.com/search) although my hunch is that this would be something we'd want to do some light usability testing on, which LP is right in that we don't really have capacity for right now.

**Comment by jcq** [comment]
@carodew Yeah, that's the correct screen. 

One thing that seems weird to me from a UX point of view is that we currently keep the previous search results below after running a new search that doesn't return anything. We display a message stating that no cases were found for the given search, but it seems confusing that cases are still displayed below. 

Would it make more sense to replace the items in Redux whenever a search completes, even if that has no results?

Also, what do we think for copy changes? I currently have this for placeholder:
`Enter a Veteran ID, SSN, or docket number`
and this for instruction:
`Please enter a valid claims file number, SSN, or docket number to search for all available cases.`

Do we need to somehow indicate in the copy that it has to be an AMA docket number?

**Comment by carodew** [comment]
I replicated what is happening in demo, and I agree it's wonky to keep displaying the old results. 

I think we'd want a designer to come up with a mock up for how we want the screen to look when there are no results though, rather than just having it blank below. The current handling of just the error message at top seems pretty wonky. I don't think it would be a big effort for design, but it wouldn't be none effort, so this might be something we need to figure out how to prioritize? Thoughts, @jimruggiero ?  

![image](https://user-images.githubusercontent.com/10144074/62785680-237d1380-ba8f-11e9-8d6f-70ce05bece52.png)

**Comment by carodew** [comment]
As far as the language for the instructional text, I think one principle we'd like to follow more consistently is 'teach the process as you go" – aka, users learn about how Caseflow and the process works as they're using the app, so I would err on the side of being specific about AMA docket numbers.

**Comment by laurjpeterson** [comment]
Will create another ticket to improve search results UX.
